ui-tests/src/test/java/org/openmrs/reference/AddDiagnosisToVisitNoteTest.java

```diff
@@ -10,34 +10,56 @@
 package org.openmrs.reference;
 
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
 
+import org.junit.Before;
 import org.junit.Test;
-import org.junit.Ignore;
 import org.junit.experimental.categories.Category;
 import org.openmrs.reference.groups.BuildTests;
+import org.openmrs.reference.helper.TestPatient;
+import org.openmrs.reference.page.ActiveVisitsPage;
 import org.openmrs.reference.page.ClinicianFacingPatientDashboardPage;
+import org.openmrs.reference.page.FindPatientPage;
+import org.openmrs.reference.page.PatientVisitsDashboardPage;
+import org.openmrs.reference.page.RegistrationPage;
 import org.openmrs.reference.page.VisitNotePage;
 
 
 public class AddDiagnosisToVisitNoteTest extends ReferenceApplicationTestBase {
     private ClinicianFacingPatientDashboardPage patientDashboardPage;
     private VisitNotePage visitNotePage;
+    private ActiveVisitsPage activeVisits;
+    private FindPatientPage findPatientPage;
+    private RegistrationPage registrationPage;
+    private TestPatient patient;
+    private PatientVisitsDashboardPage patientVisitsDashboard;
     
-    @Test
-    @Ignore //unstable
+    /* To make this test more stable, a new patient is registered, then given an active visit
+     * The test is then performed on this new patient
+     */ 
+    @Before
+    public void setUp() throws InterruptedException {
+    	registrationPage = homePage.openRegisterAPatientApp();
+    	patient = new TestPatient();
+    	registrationPage.enterGenericPatient(patient);
+    	patientDashboardPage = registrationPage.confirmPatient();
+    	patientDashboardPage.goToHome();
+    	findPatientPage = homePage.clickOnFindPatientRecord();
+    	patientDashboardPage = findPatientPage.clickOnFirstPatient();
+    	patientVisitsDashboard = patientDashboardPage.startVisit();
+    	patientVisitsDashboard.goToHome();
+    }
+    
+    @Test // RA-1117
     @Category(BuildTests.class)
-    public void AddDiagnosisToVisitNoteTest() throws Exception {
-    	
-        patientDashboardPage = homePage.goToActiveVisitPatient();
+    public void AddDiagnosisVisitNoteTest() throws InterruptedException {
+        activeVisits = homePage.goToActiveVisitsSearch();
+        patientDashboardPage = activeVisits.goToPatientDashboardOfLastActiveVisit();
         visitNotePage = patientDashboardPage.goToVisitNote();
         visitNotePage.enterDiagnosis("Pne");
         visitNotePage.enterSecondaryDiagnosis("Bleed");
         assertEquals("Pneumonia", visitNotePage.primaryDiagnosis());
         assertEquals("Bleeding", visitNotePage.secondaryDiagnosis());
         visitNotePage.save();
-        assertNotNull(patientDashboardPage.visitLink());
```

Comment:
I took it out when I was trying to track down an error. I've added it back in. Thanks for pointing it out!